### ISSUE TITLE
I can’t reproduce the performance of PN

### ISSUE THREAD

**Issue by skingorz** [Frankluox/FewShotPL#2]
I can’t reproduce the performance of PN. In your readme，the accuracy of PN can reach  61.09. However only 59.26 on test set I I can reproduce while  62.31 on val set. I train it on 3090*2 and use the same config  as set_config_PN.py. Are there any detiles I haven't noticed?

**Comment by Frankluox** [comment]
I will run the code to see if there exists a problem. Please wait.

**Comment by Frankluox** [comment]
I have checked and ensured that the accuracy can be achieved.  One thing to notice is that there is only one trained 5-shot model, testing for both 1-shot and 5-shot settings. Did you run a new 1-shot model for 1-shot testing? This will likely harm the performance. And make sure you are using the best model checkpoint with the highest validation accuracy, instead of the last one.

**Comment by Frankluox** [comment]
The optimal choice of the total training batch size is 2.

**Comment by skingorz** [comment]
> I have checked and ensured that the accuracy can be achieved. One thing to notice is that there is only one trained 5-shot model, testing for both 1-shot and 5-shot settings. Did you run a new 1-shot model for 1-shot testing? This will likely harm the performance. And make sure you are using the best model checkpoint with the highest validation accuracy, instead of the last one.

I did train the 1-shot code for 1-shot training. I will try a 5-shot training. Thank you!
